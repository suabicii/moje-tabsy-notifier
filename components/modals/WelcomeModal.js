import React, {useState} from "react";
import {Modal, Pressable, Text, View, StyleSheet} from "react-native";

function WelcomeModal({isVisible = false}) {
    const [modalVisible, setModalVisible] = useState(isVisible);

    return (
        <View style={styles.centeredView}>
            <Modal
                animationType="fade"
                transparent={true}
                visible={modalVisible}
                onRequestClose={() => setModalVisible(!modalVisible)}
            >
                <View style={styles.centeredView}>
                    <View style={styles.modalView}>
                        <Text style={styles.modalText}>
                            Od teraz bÄ™dziesz otrzymywaÅ‚/-a powiadomienia ðŸ””,
                            ktÃ³re bÄ™dÄ… Ci przypominaÄ‡ o zaÅ¼yciu lekÃ³w/suplementÃ³w ðŸ’Š
                            lub koÅ„czÄ…cych siÄ™ zapasach
                        </Text>
                        <Text style={[styles.modalText, styles.modalTextSecondary]}>
                            W godzinie zaÅ¼ycia medykamentu pojawi siÄ™ przycisk,
                            dziÄ™ki ktÃ³remu potwierdzisz, Å¼e wÅ‚aÅ›nie zostaÅ‚
                            przez Ciebie przyjÄ™ty âœ…
                        </Text>
                        <Pressable
                            style={[styles.button, styles.buttonClose]}
                            onPress={() => setModalVisible(!modalVisible)}
                        >
                            <Text style={styles.textStyle}>OK</Text>
                        </Pressable>
                    </View>
                </View>
            </Modal>
        </View>
    );
}

const styles = StyleSheet.create({
    centeredView: {
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
        marginTop: 22
    },
    modalView: {
        margin: 20,
        backgroundColor: "white",
        borderRadius: 20,
        padding: 35,
        alignItems: "center",
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2
        },
        shadowOpacity: 0.25,
        shadowRadius: 4,
        elevation: 5
    },
    button: {
        borderRadius: 25,
        padding: 15,
        elevation: 2
    },
    buttonClose: {
        backgroundColor: "#2196F3",
    },
    textStyle: {
        color: "white",
        fontWeight: "bold",
        textAlign: "center"
    },
    modalText: {
        fontSize: 22,
        marginBottom: 15,
        textAlign: "center"
    },
    modalTextSecondary: {
        color: "#525252",
        fontSize: 20
    }
});

export default WelcomeModal;