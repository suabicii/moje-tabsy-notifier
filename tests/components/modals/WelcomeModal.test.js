/**
 * @jest-environment jsdom
 * */
import React from "react";
import renderer from "react-test-renderer";
import WelcomeModal from "../../../components/modals/WelcomeModal";
import {render, act, fireEvent, screen} from "@testing-library/react-native";
import AsyncStorage from "@react-native-async-storage/async-storage";

it('should correctly render WelcomeModal', () => {
    const tree = renderer.create(<WelcomeModal isVisible={true}/>).toJSON();

    expect(tree).toMatchSnapshot();
});

it('should save information about welcome message disability in AsyncStorage if checkbox is checked', async () => {
    render(<WelcomeModal isVisible={true}/>);

    await act(async () => {
        fireEvent.press(screen.getByTestId('welcomeMsgDisableCheckbox'));
    });
    const msgDisable = await AsyncStorage.getItem('welcome_msg_disable');

    expect(msgDisable).toBeTruthy();
});

it('should remove information about welcome message disability from AsyncStorage if checkbox is unchecked', async () => {
    render(<WelcomeModal isVisible={true}/>);

    await act(async () => {
        fireEvent.press(screen.getByTestId('welcomeMsgDisableCheckbox')); // check
    });
    await act(async () => {
        fireEvent.press(screen.getByTestId('welcomeMsgDisableCheckbox')); // uncheck
    });
    const msgDisable = await AsyncStorage.getItem('welcome_msg_disable');

    expect(msgDisable).toBeFalsy();
});