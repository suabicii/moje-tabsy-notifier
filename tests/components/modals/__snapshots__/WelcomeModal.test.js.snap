// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render WelcomeModal 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 22,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    testID="welcomeModal"
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 22,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "elevation": 5,
            "margin": 20,
            "padding": 35,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 22,
              "marginBottom": 15,
              "textAlign": "center",
            }
          }
        >
          Od teraz bÄ™dziesz otrzymywaÅ‚/-a powiadomienia ğŸ””, ktÃ³re bÄ™dÄ… Ci przypominaÄ‡ o zaÅ¼yciu lekÃ³w/suplementÃ³w ğŸ’Š lub koÅ„czÄ…cych siÄ™ zapasach
        </Text>
        <Text
          style={
            [
              {
                "fontSize": 22,
                "marginBottom": 15,
                "textAlign": "center",
              },
              {
                "color": "#525252",
                "fontSize": 20,
              },
            ]
          }
        >
          W godzinie zaÅ¼ycia medykamentu pojawi siÄ™ przycisk, dziÄ™ki ktÃ³remu potwierdzisz, Å¼e wÅ‚aÅ›nie zostaÅ‚ przez Ciebie przyjÄ™ty âœ…
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          fillColor="#78c2ad"
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
              },
              {
                "marginBottom": 15,
              },
            ]
          }
          testID="welcomeMsgDisableCheckbox"
          text="Nie pokazuj mi tego wiÄ™cej"
          textStyle={
            {
              "textDecorationLine": "none",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 12.5,
                "height": 25,
                "justifyContent": "center",
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
                "width": 25,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "borderColor": "#78c2ad",
                    "borderRadius": 12.5,
                    "borderWidth": 1,
                    "height": 25,
                    "justifyContent": "center",
                    "width": 25,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "marginLeft": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#757575",
                    "fontSize": 16,
                    "textDecorationLine": "none",
                  },
                  {
                    "textDecorationLine": "none",
                  },
                ]
              }
            >
              Nie pokazuj mi tego wiÄ™cej
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "borderRadius": 25,
                "elevation": 2,
                "padding": 15,
              },
              {
                "backgroundColor": "#2196F3",
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            OK
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
